app/models/vm_or_template.rb

```diff
@@ -1883,7 +1883,13 @@ def supports_snapshots?
   end
 
   def self.batch_operation_supported?(operation, ids)
-    VmOrTemplate.where(:id => ids).all? { |v| v.public_send("validate_#{operation}")[:available] }
+    VmOrTemplate.where(:id => ids).all? do |s|
```

Comment:
I would always try to avoid one letter variables. In this case `do |vm_or_template|` would be better.
Although for such a narrow scope its ok.
And it will be touched again once we removed all `validate_` methods.

so you can keep it if you like